packages/babel-plugin-syntax-record-and-tuple/src/index.js

```diff
@@ -0,0 +1,19 @@
+import { declare } from "@babel/helper-plugin-utils";
+
+export default declare((api, options) => {
+  api.assertVersion(7);
+
+  return {
+    name: "syntax-record-and-tuple",
+
+    manipulateOptions(opts, parserOpts) {
+      parserOpts.plugins.push([
+        "recordAndTuple",
+        {
+          hash: options.hash,
```

Comment:
I prefer to expose `syntaxType` to the user land, because
- When either syntax variants is accepted, we can make that as default. We have similar [design](https://babeljs.io/docs/en/next/babel-plugin-proposal-pipeline-operator.html#via-babelrc-recommended) on pipeline-operator. 
- We don't have to process option precedence issues like `hash: false, bar: false` and `hash: true, bar: true`.
- More open to competing proposals, if any